server/controllers/categorised/employees.js

```diff
@@ -21,58 +51,208 @@ exports.list = function (req, res, next) {
+  'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+  'FROM employee ' +
+  ' JOIN grade ON employee.grade_id = grade.uuid ' +
+  ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+  ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+  ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+  ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+  ' LEFT JOIN service ON service.id = employee.service_id ' +
+  'WHERE employee.id = ? ';
+
+  db.exec(sql, [req.params.id])
+  .then(function (rows) {
+    if (!rows.length) {
+      throw codes.ERR_NOT_FOUND;
+    }
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+};
+
+/**
+* Update details of an employee referenced by an `id` in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // PUT /employees/:id : Update details of an employee
+* var employees = require('categorised/employees');
+* employees.update(request, response, next);
+*/
+exports.update = function (req, res, next) {
+  var sql = 'UPDATE employee SET ? WHERE employee.id = ?';
+
+  db.exec(sql, [req.body, req.params.id])
+  .then(function (row) {
+    if (!row.affectedRows) { throw codes.ERR_NOT_FOUND; }
+
+    var sql2 =
+    'SELECT ' +
+    'employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
+    'employee.postnom, employee.sexe, employee.dob, employee.date_embauche, employee.service_id, ' +
+    'employee.nb_spouse, employee.nb_enfant, employee.grade_id, employee.locked, grade.text, grade.basic_salary, ' +
+    'fonction.id AS fonction_id, fonction.fonction_txt, service.name AS service_txt, ' +
+    'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, employee.daily_salary, employee.location_id, ' +
+    'grade.code AS code_grade, debitor.uuid as debitor_uuid, debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
+    'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+    'FROM employee ' +
+    ' JOIN grade ON employee.grade_id = grade.uuid ' +
+    ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+    ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+    ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+    ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+    ' LEFT JOIN service ON service.id = employee.service_id ' +
+    'WHERE employee.id = ? ';
+
+    return db.exec(sql2, [req.params.id]);
+  })
+  .then(function (rows) {
+    if (!rows.length) {
+      throw codes.ERR_NOT_FOUND;
+    }
+    // returing date in 'yyyy-mm-dd' format
+    rows[0].dob = util.toMysqlDate(rows[0].dob);
```

Comment:
Thank you Jon, i try it...

2016-01-12 9:15 GMT+01:00 Jonathan Niles <notifications@github.com>:

> In server/controllers/categorised/employees.js
> <https://github.com/IMA-WorldHealth/bhima-2.X/pull/12#discussion_r49425309>
> :
>
> > +    ' JOIN grade ON employee.grade_id = grade.uuid ' +
> > +    ' JOIN fonction ON employee.fonction_id = fonction.id ' +
> > +    ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
> > +    ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
> > +    ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
> > +    ' LEFT JOIN service ON service.id = employee.service_id ' +
> > +    'WHERE employee.id = ? ';
> > +
> > +    return db.exec(sql2, [req.params.id]);
> > +  })
> > +  .then(function (rows) {
> > +    if (!rows.length) {
> > +      throw codes.ERR_NOT_FOUND;
> > +    }
> > +    // returing date in 'yyyy-mm-dd' format
> > +    rows[0].dob = util.toMysqlDate(rows[0].dob);
>
> Hmm... This solution works, but doesn't scale well and has maintainability
> issues.
>
> I wonder if we used "DATETIME" instead of "DATE" in our column definitions
> if that would solve the issues we are having with dates. Could you
> experiment with this and see what happens? I'd really like to see an
> integration test that does something like this:
>
> var createdAt = new Date('2015-03-02 15:30:20');var employee = { /* properties */, dob : createdAt, /* more properties */ };
> // inside a testit('preserves the correct timestamp in the database', function () {
>   return agent.post('/employees/')
>   .send(employee)
>   .then(function (res) {
>     return agent.get('/employees/' + res.body.id);
>   })
>   .then(function (res) {
>     var dbEmployee = res.body;
>     expect(dbEmployee.dob).to.eql(createdAt);
>     // etc...
>   });
> });
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IMA-WorldHealth/bhima-2.X/pull/12/files#r49425309>.
>